### DISCUSSION TITLE
When saving screenshots, some settings don't take effect

### DISCUSSION THREAD

**Discussion by jromazanov** [pyvista/pyvistaqt#325] [Q&A]
I'd like to make a 3D mesh plot, define a camera rotation in the script, show an interactive window with some adjusted settings (e.g. camera rotation, canvas background color), and finally save a screenshot into `test.png`.

This works fine with pyvista's `Plotter` class, but apparently not with pyvistaqt's `BackgroundPlotter` class.
Below is a minimal example. If I run with `plotType = 'pv'`, it works fine, the screenshot has the same rotation and background color as the interactive window. But if I run with `plotType = 'pvqt'`, the screenshot forgets the camera rotation and always uses the same default rotation. 

Question: Is there a different way I need to pass the settings and save the screenshot? Or is this a bug?

Expected behavior: (as seen in interactive window)
<img src="https://user-images.githubusercontent.com/20295983/223051588-10c77a61-c216-4ae1-bda0-f3239821d16a.png" width="300">

Actual behavior: (as produced in screenshot)
<img src="https://user-images.githubusercontent.com/20295983/223051498-481100e7-91a4-4c28-a35a-eaea4459eabb.png" width="300">

(Interestingly, other settings are transferred correctly to the pyvistaqt screenshot, e.g. `pl.view_yz()` would take effect in both interactive window and screenshot.)

```
import pyvista as pv
import pyvistaqt as pvqt

# uncomment one:
#plotType = 'pv'
plotType = 'pvqt'

# create plotter
if plotType == 'pv':
    pl = pv.Plotter()
else:
    pl = pvqt.BackgroundPlotter()

# do plot
cube = pv.Cube()
pl.add_mesh(cube, show_edges=True)

# change camera angle
pl.camera.azimuth = 45.0

# change background color
pl.set_background('white')

# show figure and take screenshot
if plotType == 'pv':
    pl.show(screenshot='test.png')
else:
    screenshot = pl.screenshot('test.png')
    pl.app.exec_()
```

**Comment by larsoner** [comment]
Can you see if some `pl.app.processEvents()` calls before your `pl.screenshot` fix things? I'm guessing this is an event-driven problem, assuming what you actually see in the window after / during the `app.exec_()` is the correct thing (i.e., it is correct but the screenshot is not).

**Reply by jromazanov** [reply]
Inserting `pl.app.processEvents()` before `pl.screenshot` seems to have no effect in my case. 
In case it's relevant: I'm running on Ubuntu 22.04 with Python 3.10.6, pyvista 0.37.0, pyvistaqt 0.9.0, OpenGL 4.6, vtk 9.2.2, QtPy 2.0.0 (PyQt5=5.15.2).
And yes, the window / `app.exec_()` shows the correct thing, but the screenshot doesn't. I've edited the question to include screenshots.

**Reply by adeak** [reply]
I can repro your issue @jromazanov, and I can even see a brief flash of the wrong render before the final one shows interactively. Adding `pl.render()` before taking the screenshot fixes it. I don't know what's going on though.

**Reply by jromazanov** [reply]
Thank you @adeak, I confirm `pl.render()` fixes it for me!

**Reply by josephko91** [reply]
I had this exact same issue! `pl.render()` also worked for me.
